blog/content/post/2020-07-21-secure-app-coding-practices.md

```diff
@@ -0,0 +1,74 @@
+First of all it is good practice to use separate channels of communication for sharing sensitive information such as a PIN or password. For example, you can use an HTTPS network connection to share encrypted data between the client and server, and then use APNS/GCM or SMS to share the PIN or token/key with the client app. That way, even if one channel of communication is compromised, the security of the overall system remains intact. Additionaly most local development and hosting environments now provide a mechanism for registering custom [environment variables](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.1&tabs=windows/). 
+
+Don't forget though that keeping your server code on the repo, even without passwords, can be risky. If your script has any vulnerabilities, the hacker will be able to easily detect them, and then attack the application published, for example to download data from the database using it.
+
+
+### Strong password
+Hiding your password won't do anything if it's just weak. Remember that hackers often operate on powerful hardware that makes code-cracking applications extremely efficient. How can we protect ourselves against it? The attack called "by dictionary" tries the most popular combinations from words that already exist. That can be phrases or words from the dictionary.
+The most efficient way to fight against this attack is to use a password containing at least 14 characters that includes special characters, numbers, letters lower and upper cases. It is also very important to not use the same password for everything, they must be all different. It’s up to you find a proper way to remember them all... This is of course a very general answer. However, this is the minimum that we should stick to. It is difficult to create an exact list of requirements, as these often change, e.g. until recently it was a very popular practice to force users to periodically change their password. Meanwhile, Microsoft says, based on its experience, that this approach does more harm than good. Out of laziness, users choose almost identical passwords anyway, changing only one of the last characters, which is why, for example, in their [new guidelines](https://docs.microsoft.com/en-us/microsoft-365/admin/misc/password-policy-recommendations?view=o365-worldwide/) for Office 365, the technological giant recommends abandoning this idea. 
+
+### Always validate the data also on the server
+
+The "required" attribute for the text field or "disabled" for the select item, we are able to turn off these items in the development tools. Validations written in JS can also be modified on the fly, or you can simply turn off JavaScript.
```

Comment:
We could mention the most common vulnerabilities, like SQL Injection or cross-origin request forgery, and how one should treat every user input as being potentially harmful.

Reply:
I would like to address the issue of injections, neutralizing threats, "broken authentication" and testing  in another article because the topic is really extensive. Here I only focused on the basic practices of secure coding